File: ts-sdk/src/sdk/models/shared/transactiondata.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { Posting } from "./posting";


export class TransactionData extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=metadata" })
  metadata?: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=postings", elemType: Posting })
  postings: Posting[];

  @SpeakeasyMetadata({ data: "json, name=reference" })
  reference?: string;

  @SpeakeasyMetadata({ data: "json, name=timestamp" })
  timestamp?: Date;
}

File: go-sdk/pkg/models/shared/transactiondata.go
package shared

import (
	"time"
)

type TransactionData struct {
	Metadata  map[string]interface{} `json:"metadata,omitempty"`
	Postings  []Posting              `json:"postings"`
	Reference *string                `json:"reference,omitempty"`
	Timestamp *time.Time             `json:"timestamp,omitempty"`
}
